pkg/machine/qemu/machine.go

```diff
@@ -1287,23 +1334,74 @@ func (v *MachineVM) waitAPIAndPrintInfo(forwardState apiForwardingState, forward
 	}
 }
 
-func (v *MachineVM) writeConfig() error {
-	// GetConfDir creates the directory so no need to check for
-	// its existence
-	vmConfigDir, err := machine.GetConfDir(vmtype)
+// update returns the content of the VM's
+// configuration file in json
+func (v *MachineVM) update() error {
+	if err := v.setConfigPath(); err != nil {
+		return err
+	}
+	b, err := v.ConfigPath.Read()
 	if err != nil {
 		return err
 	}
+	if os.IsNotExist(err) {
```

Comment:
This will never be true, because you just returned the err in the line above.  The lines above should probably be removed.